service/src/main/java/bio/terra/workspace/db/DbRetryUtils.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.workspace.db;
+
+import bio.terra.common.db.DatabaseRetryUtils;
+import java.time.Duration;
+import java.util.List;
+import org.springframework.dao.RecoverableDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+/**
+ * This class contains wrappers around JDBC methods which run them with retries. Only DB exceptions
+ * considered retryable ({@link RecoverableDataAccessException} and {@link
+ * org.springframework.dao.TransientDataAccessException}) are retried. See Spring documentation of
+ * these exceptions for more information:
+ * https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/DataAccessException.html
+ *
+ * <p>Retries are with no delay, up to a certain maximum number of retries.
+ */
+public class DbRetryUtils {
```

Comment:
Why not put all these in the common lib?

Reply:
I think we'll want to call `executeAndRetry` with different parameters in different services - I think the 0 time retries work here, but would cause problems for slow queries, where the conflict might not be immediately over. Having the base `executeAndRetry` in common lib and calling with different parameters from each service seemed like the right compromise to me

Reply:
This could be done in common lib by having a setter for default backoff and attempts, and then a variant of `executeAndRetry` that did not take those arguments.

Maybe collect these suggestions in a follow-on PR. Along with different timings for transaction vs connection failures.

Reply:
Sounds good, PF-805 to track these suggestions